### ISSUE TITLE
Stuck on downloading...

### ISSUE THREAD

**Issue by Zyles** [composer/composer#9358]
```
$ php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
$ php -r "if (hash_file('sha384', 'composer-setup.php') === 'c31c1e292ad7be5f49291169c0ac8f683499edddcfd4e42232982d0fd193004208a58ff6f353fde0012d35fdd72bc394') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
$ php composer-setup.php
All settings correct for using Composer
Downloading...
```

**Comment by Zyles** [comment]
Works now, apparently took 10 minutes to install!

**Comment by Zyles** [comment]
No not really. It is a VPS server. 40 Gbps Network In 6000 Mbps Network Out

Could be a memory issue, only have 8GB on it.

**Comment by Seldaek** [comment]
That script is downloading a 2MB phar file, I don't think that should use a lot of RAM :)

```
$ time php composer-setup.php
All settings correct for using Composer
Downloading...

Composer (version 2.0.2) successfully installed to: /var/www/composer/testb/composer.phar
Use it: php composer.phar

real    0m0.427s
user    0m0.069s
sys     0m0.020s
```

Weird.. but hopefully it was a one off glitch

**Comment by fgiongo** [comment]
Strange, I experienced the same issue, but it was less than 10 minutes.

**Comment by jaideep-recruitcrm** [comment]
Having the same issue right now

**Comment by chibyk101** [comment]
i'm having same issue right now, i thought it was my internet connection, I noticed every other thing I tried downloading using curl was blazing fast.

it's clearly not an internet connection issue

**Comment by Iscrimou** [comment]
I'm having same issue. Following installation instruction from official documentation on WSL2

**Comment by chibyk101** [comment]
> I'm having same issue. Following installation instruction from official documentation on WSL2

Check here https://getcomposer.org/doc/articles/troubleshooting.md#operation-timed-out-ipv6-issues-

It solved the issue for me. The issue was IPv6 taking precedence over IPv4

**Comment by RandalVanheede** [comment]
For anyone reading this, I got this issue and it WAS solved by telling Linux to prefer IPv4 over IPv6 as explained in the link above, BUT! After successfully installing Composer, I ran `composer diagnose` and it told me this:

```curl version: missing, using php streams fallback, which reduces performance```

I then proceeded to run `sudo apt install php8.4-curl` (replace 8.4 with your php version), and I removed the line that was previously added to `/etc/gai.conf` removing Linux's preference for IPv4, and everything runs smoothly now.


TL;DR: install the curl extension for PHP.
